server/app/artist/serializer.py

```diff
@@ -0,0 +1,59 @@
+"""
+Serializers for artist APIs
+"""
+from rest_framework import serializers
+
+from core.models import Artist, Track
+
+
+class TrackSerializer(serializers.ModelSerializer):
+    """Serializer for tracks."""
+
+    class Meta:
+        model = Track
+        fields = ['id', 'title']
+        read_only_fields = ['id']
+
+class ArtistSerializer(serializers.ModelSerializer):
+    """Serializer for artists."""
+    tracks = TrackSerializer(many=True, required=False)
+
+    class Meta:
+        model = Artist
+        fields = ['id', 'first_name', 'last_name', 'tracks']
+        read_only_fields = ['id']
+
+    def _get_or_create_tracks(self, tracks, artist):
+        """Handle getting or creating tracks"""
+        for track in tracks:
+            track_obj, _ = Track.objects.get_or_create(**track)
+            artist.tracks.add(track_obj)
+
+    def create(self, validated_data):
+        """Create an artist"""
+        tracks = validated_data.pop('tracks', [])
+        artist = Artist.objects.create(**validated_data)
+        self._get_or_create_tracks(tracks, artist)
+        return artist
+
+    def update(self, instance, validated_data):
+        """Update recipe"""
+        tracks = validated_data.pop('tracks', None)
+        if tracks is not None:
+            instance.tracks.clear()
```

Comment:
`clear` will not delete the previous tracks. It only dissociates them from the artist. Should we delete the previous tracks too? Otherwise it seems they will stay in the database and be inaccessible